File: algorithms/2_medium/A/01936_add_minimum_number_of_rungs/solution.c
int addRungs(const int *restrict rungs, const int len, const int dist) {
    int curr = 0;
    int cnt = 0;
    for (int i = 0; i < len; i += 1) {
        const int next = rungs[i];
        const int width = next - curr;
        cnt += (width - 1) / dist;
        curr = next;
    }
    return cnt;
}

File: algorithms/2_medium/A/01936_add_minimum_number_of_rungs/solution.rs
impl Solution {
    pub fn add_rungs(rungs: Vec<i32>, dist: i32) -> i32 {
        let mut curr: i32 = 0;
        let mut cnt: i32 = 0;
        for next in rungs {
            let width = next - curr;
            cnt += (width - 1) / dist;
            curr = next;
        }
        cnt
    }
}
